test/parallel/test-single-executable-blob-config-errors.js

```diff
@@ -0,0 +1,204 @@
+    ['--experimental-sea-config', config], {
+      cwd: tmpdir.path,
+    });
+  const stderr = child.stderr.toString();
+  assert.strictEqual(child.status, 1);
+  assert(
+    stderr.includes(
+      `"main" field of ${config} is not a string`
+    )
+  );
+}
+
+{
+  tmpdir.refresh();
+  const config = join(tmpdir.path, 'no-main.json');
+  writeFileSync(config, '{"output": "test.blob"}', 'utf8');
+  const child = spawnSync(
+    process.execPath,
+    ['--experimental-sea-config', config], {
+      cwd: tmpdir.path,
+    });
+  const stderr = child.stderr.toString();
+  assert.strictEqual(child.status, 1);
+  assert(
+    stderr.includes(
+      `"main" field of ${config} is not a string`
+    )
+  );
+}
+
+{
+  tmpdir.refresh();
+  const config = join(tmpdir.path, 'no-output.json');
+  writeFileSync(config, '{"main": "bundle.js"}', 'utf8');
+  const child = spawnSync(
+    process.execPath,
+    ['--experimental-sea-config', config], {
+      cwd: tmpdir.path,
+    });
+  const stderr = child.stderr.toString();
+  assert.strictEqual(child.status, 1);
+  assert(
+    stderr.includes(
+      `"output" field of ${config} is not a string`
+    )
+  );
+}
+
+{
+  tmpdir.refresh();
+  const config = join(tmpdir.path, 'nonexistent-main-relative.json');
+  writeFileSync(config, '{"main": "bundle.js", "output": "sea.blob"}', 'utf8');
+  const child = spawnSync(
+    process.execPath,
+    ['--experimental-sea-config', config], {
+      cwd: tmpdir.path,
+    });
+  const stderr = child.stderr.toString();
+  assert.strictEqual(child.status, 1);
+  assert.match(stderr, /Cannot read main script bundle\.js/);
+}
+
+{
+  tmpdir.refresh();
+  const config = join(tmpdir.path, 'nonexistent-main-absolute.json');
+  const main = join(tmpdir.path, 'bundle.js');
+  const configJson = JSON.stringify({
+    main,
+    output: 'sea.blob'
+  });
+  writeFileSync(config, configJson, 'utf8');
+  const child = spawnSync(
+    process.execPath,
+    ['--experimental-sea-config', config], {
+      cwd: tmpdir.path,
+    });
+
+  const stderr = child.stderr.toString();
+  assert.strictEqual(child.status, 1);
+  assert(
+    stderr.includes(
+      `Cannot read main script ${main}`
+    )
+  );
+}
+
+{
+  tmpdir.refresh();
+  const config = join(tmpdir.path, 'output-is-dir-absolute.json');
+  const main = join(tmpdir.path, 'bundle.js');
+  const output = join(tmpdir.path, 'output-dir');
+  mkdirSync(output);
+  writeFileSync(main, 'console.log("hello")', 'utf-8');
+  const configJson = JSON.stringify({
+    main,
+    output,
+  });
+  writeFileSync(config, configJson, 'utf8');
+  const child = spawnSync(
+    process.execPath,
+    ['--experimental-sea-config', config], {
+      cwd: tmpdir.path,
+    });
+
+  const stderr = child.stderr.toString();
+  assert.strictEqual(child.status, 1);
+  assert(
+    stderr.includes(
+      `Cannot write output to ${output}`
+    )
+  );
+}
+
+{
+  tmpdir.refresh();
+  const config = join(tmpdir.path, 'output-is-dir-relative.json');
```

Comment:
```suggestion
  const config = 'output-is-dir-relative.json';
```

Reply:
This would not work because we are writing the config using the current directory of the test runner. The output dir is hard-coded into the JSON as relative anyway so this does not matter. Same as below.